spring-kafka/src/main/java/org/springframework/kafka/support/TopicPartitionCurrentOffset.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.support;
+
+import java.util.Objects;
+
+import org.apache.kafka.common.TopicPartition;
+
+/**
+ * Identifies the current position of a TopicPartition.
+ *
+ * @author Artem Bilan
```

Comment:
This is your class and `@since 1.1` 
:smile: 

Although I still try to understand if we need it...
Maybe simple `Map<TopicPartition, Long>` could be enough?